lib/streamlit/error_util.py

```diff
@@ -25,6 +28,29 @@
 # separator
 _streamlit_dir = os.path.join(os.path.realpath(_streamlit_dir), "")
 
+# When client.showTracebacks is False, we show a generic warning in the
+# frontend when we encounter an uncaught app exception.
+_GENERIC_UNCAUGHT_EXCEPTION_TEXT = (
+    "Whoops - something went wrong! An error has been logged."
+)
+
+
+def handle_uncaught_app_exception(e: BaseException) -> None:
+    """Handle an exception that originated from a user app.
+    By default, we show exceptions directly in the browser. However,
+    if the user has disabled client tracebacks, we display a generic
+    warning in the frontend instead.
+    """
+    if config.get_option("client.showTracebacks"):
+        LOGGER.debug(e)
+        st.exception(e)
+        # TODO: Clean up the stack trace, so it doesn't include ScriptRunner.
```

Comment:
(This TODO existed already - this whole branch was just moved from ScriptRunner)